visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.service.ts

```diff
@@ -0,0 +1,111 @@
+import { StoreService, StoreSubscriber } from "../../../store.service"
+import { IRootScopeService } from "angular"
+import { calculateNewEdgeMetricData, EdgeMetricDataActions } from "./edgeMetricData.actions"
+import { isActionOfType } from "../../../../util/reduxHelper"
+import {
+	AttributeTypes,
+	AttributeTypeValue,
+	BlacklistItem,
+	CodeMapNode,
+	EdgeMetricCount,
+	EdgeMetricData
+} from "../../../../codeCharta.model"
+import { FileState } from "../../../../model/files/files"
+import { HierarchyNode } from "d3"
+import { BlacklistService, BlacklistSubscriber } from "../../fileSettings/blacklist/blacklist.service"
+import { FilesSelectionSubscriber, FilesService } from "../../files/files.service"
+import { nodeEdgeMetricsMap } from "./edgeMetricData.reducer"
+import { AttributeTypesService, AttributeTypesSubscriber } from "../../fileSettings/attributeTypes/attributeTypes.service"
+
+export interface EdgeMetricDataSubscriber {
+	onEdgeMetricDataChanged(edgeMetricData: EdgeMetricData[])
+}
+
+export class EdgeMetricDataService implements StoreSubscriber, BlacklistSubscriber, FilesSelectionSubscriber, AttributeTypesSubscriber {
+	private static EDGE_METRIC_DATA_CHANGED_EVENT = "edge-metric-data-changed"
+	public static NONE_METRIC = "None"
+
+	constructor(private $rootScope: IRootScopeService, private storeService: StoreService) {
+		StoreService.subscribe(this.$rootScope, this)
+		BlacklistService.subscribe(this.$rootScope, this)
+		FilesService.subscribe(this.$rootScope, this)
+		AttributeTypesService.subscribe(this.$rootScope, this)
+	}
+
+	public onStoreChanged(actionType: string) {
+		if (isActionOfType(actionType, EdgeMetricDataActions)) {
+			this.notify(this.select())
+		}
+	}
+
+	public onBlacklistChanged(blacklist: BlacklistItem[]) {
+		this.storeService.dispatch(calculateNewEdgeMetricData(this.storeService.getState().files, blacklist))
+	}
+
+	public onFilesSelectionChanged(files: FileState[]) {
+		this.storeService.dispatch(calculateNewEdgeMetricData(files, this.storeService.getState().fileSettings.blacklist))
+	}
+
+	public onAttributeTypesChanged(attributeTypes: AttributeTypes) {
```

Comment:
The `attributeTypes` are never used?

```suggestion
	public onAttributeTypesChanged() {
```

Reply:
The function is part of an interface (in this case `AttributeTypesSubscriber`). I didn't know that you could simply removed parameters from the implemented function.
What is a good practice here? Keep the parameters of the function, even though they might not be used or remove them?

Reply:
Removing them is fine.

The example about removing the first argument is not comparable.

Reply:
This has been adressed in #1156